src/core/splitter/EdgeGraph.js

```diff
@@ -0,0 +1,688 @@
+
+					const insertedEdge = edgePool.getInstance();
+					insertedEdge.start.copy( point );
+					insertedEdge.startIndex = index;
+					insertedEdge.end.copy( triangle.points[ nextEdgeIndex ] );
+					insertedEdge.endIndex = triangle.getVertexIndex( nextEdgeIndex );
+
+					const finalEdgeIndex0 = ( edgeIndex + 1 ) % 3;
+					const newTri0 = trianglePool.getInstance();
+					newTri0.setEdge( 0, l0, true );
+					newTri0.setEdge( 1, triangle.edges[ finalEdgeIndex0 ].edge, triangle.edges[ finalEdgeIndex0 ].reversed );
+					newTri0.setEdge( 2, insertedEdge, true );
+
+					const finalEdgeIndex1 = ( edgeIndex + 2 ) % 3;
+					const newTri1 = trianglePool.getInstance();
+					newTri1.setEdge( 0, l1, true );
+					newTri1.setEdge( 1, insertedEdge, false );
+					newTri1.setEdge( 2, triangle.edges[ finalEdgeIndex1 ].edge, triangle.edges[ finalEdgeIndex1 ].reversed );
+
+					triangles.splice( triangles.indexOf( triangle ), 1 );
+					triangles.push( newTri0, newTri1 );
+					edges.push( insertedEdge );
+
+				}
+
+				edges.push( l0, l1 );
+				edges.splice( intersectingEdge, 1 );
+
+			}
+
+		}
+
+		return index;
+
+	}
+
+	findMatchingPointIndex( p ) {
+
+		// find the matching vertex for the give point if it exists in the graph
+		const points = this.points;
+		let closestIndex = null;
+		let closestDist = Infinity;
+		for ( let i = 0, l = points.length; i < l; i ++ ) {
+
+			const d = p.distanceTo( points[ i ] );
+			if ( d < EPSILON && d < closestDist ) {
+
+				closestIndex = i;
+				closestDist = d;
+
+			}
+
+		}
+
+		return closestIndex;
+
+	}
+
+	markMatchingEdgeRequired( inserting ) {
+
+		const { edges } = this;
+
+		_edgesToSwap.length = 0;
+		for ( let i = 0, l = edges.length; i < l; i ++ ) {
+
+			// swap the edge if we don't emanate from the same point
+			const other = edges[ i ];
+			if (
+				other.startIndex !== inserting.startIndex &&
+				other.startIndex !== inserting.endIndex &&
+				other.endIndex !== inserting.startIndex &&
+				other.endIndex !== inserting.endIndex
+			) {
+
+				// TODO
+				// if ( areEdgesParallel( inserting, other ) ) {
+
+				// 	let sp = inserting.closestPointToPointParameter( other.start, false );
+				// 	let ep = inserting.closestPointToPointParameter( other.end, false );
+
+				// 	if ( ! (
+				// 		sp < 0 && ep < 0 ||
+				// 		sp > 1 && ep > 1
+				// 	) ) {
+
+				// 		// TODO: but do they overlap
+				// 		// console.log( 'THEYRE PARALLEL', sp, ep );
+
+				// 	}
+
+				// }
+
+				// TODO: make sure we account for parallel scenarios
+				if ( lineIntersect( inserting, other, _vec ) ) {
+
+					_edgesToSwap.push( other );
+
+				}
+
+			} else {
+
+				// if we found an edge that matches without swapping then theres no need
+				// to continue
+				if ( doEdgesMatch( inserting, other ) ) {
+
+					other.required = true;
+					return true;
+
+				}
+
+			}
+
+		}
+
+		// try for a few iterations to swap edges until they work
+		for ( let i = 0; i < SWAP_ITERATIONS; i ++ ) {
```

Comment:
Could you explain why 3 was chosen? I ended up with a case where I needed a few more iterations, and _edgesToSwapLength ended up being enough

Reply:
My case there ended up being 7 edges, and I needed 4 iterations.

Reply:
I think 3 is just the number I was using for my testing at the time.